src/AgentCommerce/HoneyWebhookService.php

```diff
@@ -0,0 +1,175 @@
+        }
+
+        $storefront = $productExport->getStorefrontSalesChannel();
+        if (!$storefront) {
+            throw HoneyWebhookExceptions::storefrontSalesChannelNotFound();
+        }
+
+        $route = $this->router->getRouteCollection()->get('store-api.product.export');
+        if (!$route) {
+            throw HoneyWebhookExceptions::invalidProductExportRoute();
+        }
+
+        $path = str_replace(['{accessKey}', '{fileName}'], [$productExport->getAccessKey(), $productExport->getFileName()], $route->getPath());
+        $url = $storefront->getHreflangDefaultDomain()?->getUrl() ?? $storefront->getDomains()?->first()?->getUrl();
+
+        $tokenBuilder = Builder::new(new JoseEncoder(), ChainedFormatter::default());
+
+        return $tokenBuilder
+            ->withClaim('storeName', $salesChannel->getTranslation('name'))
+            ->withClaim('storeUrl', $url)
+            ->withClaim('country', $salesChannel->getCountry()?->getIso())
+            ->withClaim('currency', $salesChannel->getCurrency()?->getIsoCode())
+            ->withClaim('favIcon', 'https://localhost/favicon.ico') // TODO: Need to be load
+            ->withClaim('shippingCountries', $storefront->getCountries()?->map(fn (CountryEntity $country) => $country->getIso()))
+            ->withClaim('paypalMerchantId', $this->credentialsUtil->getMerchantPayerId($storefront->getId()))
+            ->withClaim('shopwareMerchantId', $salesChannel->getId())
+            ->withClaim('catalogDownloadUrl', rtrim($url ?? '', '/') . $path)
+            ->getToken(new Sha256(), InMemory::plainText(random_bytes(32)))
+            ->toString();
+    }
+
+    private function loadSalesChannel(string $salesChannelId, Context $context): ?SalesChannelEntity
+    {
+        $criteria = new Criteria([$salesChannelId]);
+        $criteria->addFilter(new EqualsFilter('typeId', SwagPayPal::SALES_CHANNEL_TYPE_AGENT_COMMERCE));
+        $criteria->addAssociations([
+            'country',
+            'currency',
+            'domains',
+            'productExports.storefrontSalesChannel.domains',
+            'productExports.storefrontSalesChannel.hreflangDefaultDomain',
+            'productExports.storefrontSalesChannel.countries',
+        ]);
+
+        return $this->salesChannelRepository->search($criteria, $context)->first();
+    }
+
+    /**
+     * @throws HoneyWebhookExceptions
+     */
+    private function webhookCall(string $token, string $endpoint): HoneyWebhookResult
+    {
+        try {
+            $response = $this->client->request('POST', 'webhooks/sw/' . $endpoint, ['body' => $token]);
+        } catch (ClientException $e) {
+            $response = $e->getResponse();
+        }
+
+        $content = json_decode($response->getBody()->getContents(), true);
+        $result = new HoneyWebhookResult($content['success'] ?? !isset($content['error']), $content['message'], $content['error'] ?? null, $e ?? null);
```

Comment:
We should handle unsuccessful requests this way for a unified error and success path:
```suggestion
        if (!($content['success'] ??  true) || !isset($content['error']) {
            HoneyWebhookExceptions::apiException($content['error'] ?? 'UNKNOWN', $content['message']);
        }
```
```php
    public static function apiException(string $error, string $message): self
    {
        return new self(
            Response::HTTP_BAD_REQUEST,
            self::API_ERROR . '_' . $error,
            $message,
        );
    }
```